onnxruntime/core/providers/cuda/cuda_execution_provider.h

```diff
@@ -198,8 +198,7 @@ class CUDAExecutionProvider : public IExecutionProvider {
     CUDAGraph cuda_graph_;
     bool is_graph_captured_ = false;
     int regular_run_count_before_graph_capture_ = 0;
-    const int min_num_runs_before_cuda_graph_capture_ = 1;  // required min regular runs before graph capture for the necessary memory allocations.
-
+    const int min_num_runs_before_cuda_graph_capture_ = 2;  // required min regular runs before graph capture for the necessary memory allocations.
```

Comment:
Maybe add more extensive commentary around why the min runs needs to be 2 ?

Reply:
Alternatively, can we just keep this value as 1 and explicitly turn-off memory pattern planner is user requests usage of CUDA Graphs ? There is no real benefit of pattern planner when using CUDA Graphs.

Reply:
Turning off memory pattern is not enough. We still need 2 extra runs. The root cause might be recent change of multiple stream support, which ties Arena with stream.

A feasible solution is that we we expose Arena initial buffer size configuration in python API so that user could allocate enough Arena. But that solution requires users to know in  advance about total memory required.

It's probably fine since the first run usually take longer considering cuDNN convolution search etc.

Reply:
Not very familiar with the internals of the multi-stream support - do you see a second run allocation with memory pattern planner turned OFF && !ORT_ENABLE_STREAM disabled ?

Reply:
Disable ORT_ENABLE_STREAM flag has build error (see https://github.com/microsoft/onnxruntime/issues/16345) so I am not sure.

ORT_ENABLE_STREAM is default ON for normal builds, and we cannot disable cuda graph by default. 

I did not see other issue of using both multi-stream and cuda graph. I think it is fine since this fixes the bug of cuda graph catpure with ORT_ENABLE_STREAM and with/without memory pattern.